docs/guides/databases_with_concord.md

```diff
@@ -40,120 +161,131 @@ if (rc != SQLITE_OK) {  // Checking if something failed while opening the DB fil
+    if (sqlite3_close(db) != SQLITE_OK) {
         log_fatal("[SQLITE] Failed to close sqlite db. [%s]", sqlite3_errmsg(db));
         abort();
     }
+    return;
 }
-
-sqlite3_close(db);
 ```
 
 Pretty similar, isn't it? The difference is, the command now is "`INSERT INTO concord_guides(user_id, guild_id) values(%"PRIu64", %"PRIu64");`".
 
 The `concord_guides()` parameters **MUST match the schema for the table** that we wish to insert corresponding `values()` into.
 
-## Reading the saved records
+### Reading the saved records
 
 Finally, after saving the records, we will be reading them! This is a little more complicated than creating the table and saving records. See the following example:
 
 ```c
 sqlite3_stmt *stmt = NULL;
 
+/* Here we are setting the command, using the "SELECT ... FROM ... WHERE" command, which explains itself. */
 query = sqlite3_mprintf("SELECT user_id FROM concord_guides WHERE guild_id = %"PRIu64";", msg->guild_id);
+/* Prepares the statement and returns the status code. */
 rc = sqlite3_prepare_v2(db, query, -1, &stmt, NULL);
 
+/* Checks if anything went wrong while evaluating or executing the statement. */
 if ((rc = sqlite3_step(stmt)) != SQLITE_ERROR) {
+    /* Successfully read the record, reading it as a int64. You could use "sqlite3_column_text" to get a UTF-8 value. */
     log_trace("Sucessfully read the record: %lld.", sqlite3_column_int64(stmt, 0));
+} else {
+    /* Closing the database since an error happened. */
+    if (sqlite3_close(db) != SQLITE_OK) {
+        log_fatal("[SQLITE] Failed to close sqlite db. [%s]", sqlite3_errmsg(db));
+        abort();
+    }
 }
 
+/* Deallocating the statement struct. */
 if (sqlite3_finalize(stmt) != SQLITE_OK) {
     log_fatal("[SQLITE] Error while executing function sqlite3_finalize.");
     abort();
 }
 
+/* Closing the database. */
 sqlite3_close(db);
 ```
 
-Here we replace `sqlite3_exec` with `sqlite3_prepare_v2`, as it gives us better control over how the SQL query is executed.
-
-Now, the command is "`SELECT user_id FROM user_voice WHERE guild_id = %"PRIu64";`", where it wants to get the value of user_id, using as a search parameter the guild_id also saved before.
-
-After using `sqlite3_mprintf` to generate the query, and preparing the statement with `sqlite3_prepare_v2`, we use `sqlite3_step` for **evaluating the statement**. If it returns SQLITE_ROW, then it is going to successfully print the records with `sqlite3_column_int64` (NOTE: **You may wish to use other `sqlite3_column_*` to get other types of records, like TEXT**).
-
-## Deleting records
+### Deleting records
 
 After creating a table, saving records, and reading them, you may finally wish to delete the records.
 
 Deleting records (AKA data) is as easy (and pretty similar) to creating a table, see the following example:
 
 ```c
+/* Setting the command, using the "DELETE FROM ... WHERE" command, which deletes a row where some collumn is equal "msg->guild_id". */
```

Comment:
/* Set the command, using the "DELETE FROM ... WHERE" command, which deletes a row where some column is equal "msg->guild_id". */